### PR TITLE
Feature/sticker modifiers

### PR DESCRIPTION
**Pull request by andrewrubin**
### tl;dr
This adds the ability to modify `Sticker` images via a control button on the sticker!

#### `Stickerbook`
- Can receive a prop `stickerModifiers` which is an array of "modifier" objects of a specific shape (see the README).
- Maintains state of the currently-selected modifier as a sort of "default", so that users can select a modifier and not have to _keep_ selecting it for subsequent stickers.
- Exposes new values to the Stickerbook Provider: `stickerModifiers`, `defaultModifierIndex`, and `setDefaultModifierIndex`.

#### `Sticker`
- Accepts a prop `id` which is just a unique identifier. This can be equivalent to the `key` prop for simplicity. This prop is required if you're using modifiers.
- Accepts an `onModifierChange` callback prop. This is ultimately responsible for updating the sticker image asset based on the selected modifier.
- Adds a "modifier" control button, if `stickerModifiers` were passed to the `Stickerbook`.

#### `helpers`
- `modifyStickerImageInPlace()` abstracts away a good amount of boilerplate for updating the proper "modified" sticker asset.

### Thoughts
- This is a massive update, but probably only warrants a minor version bump.
- We'll need to test this thoroughly, so I suggest we do a beta NPM release!

### CODE REVIEW

**Comment by andrewrubin** [comment]
> Great stuff @andrewrubin ! I got a question for ya, what's the use case for maintaining a global and local `defaultIndexModifier`?

Hey @marlonmarcello !
Great question. The `global` one is just a convenience for the user. An example use case would be: you're creating a collage of all _blue_ items, but you don't want to have to toggle every single sticker to blue once you place it. The global index essentially just keeps track of what the last used modifier was, and sets that as the "default" for newly-placed stickers.

**Comment by marlonmarcello** [comment]
I see, on line `347` of the `stickers` you call the set function for the global default modifier as well every time a sticker changes its modifier, wouldn't that defeat the purpose of the example you mentioned? Couldn't the dev use the `onModifierChange` callback to keep track of the modifier instead if he wanted?

**Comment by andrewrubin** [comment]
Thanks for digging into it — don't be sorry, haha.

I chose this structure so that you don't have to worry about passing any modifier props to your sticker; something like `initialModifierIndex` for example, which would be a requirement if you're tracking the global state _solely_ on your own, as far as I can tell. That said, I went back and forth on this, and there's definitely something to what you're bringing up here. Maybe there's potential for a refactor here — I will have to get into it another day unfortunately.

Until then, have a good weekend! 👋

**Comment by marlonmarcello** [comment]
I think I got it!  
I think one thing to consider is that in the future if there is a need to add another value/state like this, might be interesting to add them all to a hook. Something like a `useStickerbook` that could take care of keeping these values and provide context the library users.

**Comment by andrewrubin** [comment]
There hasn't been any activity on this PR in ages, so I'm going to close it. I'll keep the branch around in case we do get back to the internal project we were adding this update for!
